maggy/maggy/api/routes.py

```
@@ -0,0 +1,190 @@
+    working_dir: str | None = None  # override; otherwise auto-picked
+
+
+@router.post("/execute")
+async def execute(request: Request, body: ExecuteRequest, x_api_key: str | None = Header(None)) -> dict:
+    _auth(request, x_api_key)
+    try:
+        session_id = await request.app.state.executor.start(
+            task_id=body.task_id, mode=body.mode, working_dir=body.working_dir,
+        )
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)